react/features/base/react/components/native/PagedList.js

```diff
@@ -0,0 +1,279 @@
+// @flow
+
+import React, { Component } from 'react';
+import { Text, TouchableOpacity, View } from 'react-native';
+import { connect } from 'react-redux';
+
+import { Icon } from '../../../font-icons';
+
+import styles from './styles';
+
+/**
+ * The type of the React {@code Component} props of {@link PagedList}.
+ */
+type Props = {
+
+    /**
+     * The zero-based index of the page that should be rendered (selected) by
+     * default.
+     */
+    defaultPage: number,
+
+    /**
+     * Indicates if the list is disabled or not.
+     */
+    disabled: boolean,
+
+    /**
+     * The Redux dispatch function.
+     */
+    dispatch: Function,
+
+    /**
+     * Callback to execute on page change.
+     */
+    onSelectPage: ?Function,
+
+    /**
+     * The pages of the PagedList component to be rendered.
+     *
+     * Note: An element's {@code component} may be {@code undefined} and then it
+     * won't need to be rendered.
+     */
+    pages: Array<{
+        component: ?Object,
+        icon: string | number,
+        title: string
+    }>
+};
+
+/**
+ * The type of the React {@code Component} state of {@link PagedList}.
+ */
+type State = {
+
+    /**
+     * The currently selected page.
+     */
+    pageIndex: number
+};
+
+/**
+ * A component that renders a paged list.
+ *
+ * @extends PagedList
+ */
+class PagedList extends Component<Props, State> {
+
+    /**
+     * Initializes a new {@code PagedList} instance.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            pageIndex: this._validatePageIndex(props.defaultPage)
+        };
+
+        // Bind event handlers so they are only bound once per instance.
+        this._maybeRefreshSelectedPage
+            = this._maybeRefreshSelectedPage.bind(this);
+    }
+
+    /**
+     * Renders the component.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const { disabled } = this.props;
+        const pages = this.props.pages.filter(({ component }) => component);
+
+        return (
+            <View
+                style = { [
+                    styles.pagedListContainer,
+                    disabled ? styles.pagedListContainerDisabled : null
+                ] }>
+                {
+                    pages.length > 1
+                        ? this._renderPagedList(disabled)
+                        : pages.length === 1
+                            ? React.createElement(
+
+                                // $FlowExpectedError
+                                /* type */ pages[0].component,
+                                /* props */ {
+                                    disabled,
+                                    style: styles.pagedList
+                                })
+                            : null
```

Comment:
Is there a possibility for having 0 pages? If not, can we get rid of the check then?

Reply:
It's not about having 0 pages, but about having 1 instead of 2. And it's a valid use case for Stride and similar apps that don't use calendar integration. Hence the possibility to disable that feature too.

Reply:
I wrote another answer, but later I realised what you meant. Sure, but then I'll add another check or documentation somewhere else to make sure it's clear for SDK users too